esphome/components/touchscreen/touchscreen.cpp

```diff
@@ -7,13 +7,24 @@ namespace touchscreen {
 
 static const char *const TAG = "touchscreen";
 
-void TouchscreenInterrupt::gpio_intr(TouchscreenInterrupt *store) { store->touched = true; }
+void TouchscreenInterrupt::gpio_intr(TouchscreenInterrupt *store) {
+  bool new_state = store->isr_pin_.digital_read();
+  if (new_state != store->inverted) {
+    store->touched = true;
+    if (store->component_ != nullptr) {
+      store->component_->enable_loop_soon_any_context();
+    }
+  }
```

Comment:
```suggestion
  store->touched = true;
  if (store->component_ != nullptr) {
    store->component_->enable_loop_soon_any_context();
  }
```
After checking `isr_pin_.digital_read()` handles *invert* itself